Added ability to change endpoint urls

3dgroup opened the pull request.

3dgroup committed:
Added ability to change endpoint urls

3dgroup committed:
changed opening brace to new lines

3dgroup committed:
updated closing brace position

coatesap commented:
Hi @3dgroup - from a quick skim through this change looks like it could potentially break existing installations. Is the default Realex endpoint provided somewhere?

3dgroup commented:
No your right, let me change it.

3dgroup committed:
Added Realex PayandShop.com default values

3dgroup commented:
This should have resolved the issue you mentioned by adding the default payandshop urls into the default config array.

coatesap commented on src/Omnipay/Realex/RemoteGateway.php:
```diff
@@ -25,7 +25,10 @@ public function getDefaultParameters()
+            '3dSecure'   => 0,
+            'authEndpoint'=>'https://epage.payandshop.com/epage-remote.cgi',
+            'secureDataVaultEndpoint'=>'https://epage.payandshop.com/epage-remote-plugins.cgi',
+            '3DSecureEndpoint'=>'https://epage.payandshop.com/epage-3dsecure.cgi'
```
@3dgroup - just being picky, but can you match the case of this parameter to the existing 3dSecure one - ie. lowercase "d". It's just easier for other devs to remember then if it's consistent.

coatesap commented on src/Omnipay/Realex/Message/AuthRequest.php:
```diff
@@ -10,7 +10,7 @@
 class AuthRequest extends RemoteAbstractRequest
 {
-    protected $endpoint = 'https://epage.payandshop.com/epage-remote.cgi';
+//    protected $endpoint = 'https://epage.payandshop.com/epage-remote.cgi';
```
@3dgroup - could you remove any commented-out lines if they're no longer used?

coatesap commented on src/Omnipay/Realex/Message/DeleteCardRequest.php:
```diff
@@ -71,6 +70,10 @@ protected function createResponse($data)
     public function getEndpoint()
     {
-        return $this->endpoint;
+            return $this->getParameter('AuthEndpoint');
```
@3dgroup - sorry, another picky one, but could you adjust the indentation so it follows the PSR standard of 4 spaces.

3dgroup committed:
changed formatting and removed comments

James-Byrne commented:
Hey, I just made a pull request, one part of which allows you to change one of the endpoints, is there any issues with this request as it stands? If not I will gladly take that section out of my pull request.

coatesap commented on src/Omnipay/Realex/Message/VoidRequest.php:
```diff
@@ -97,6 +95,10 @@ protected function createResponse($data)
+    }
+    public function setAuthEndpoint($value)
+    {
+        return $this->setParameter('SecureDataVaultEndpoint', $value);
```
Should getters/setters not be using lower-camel-case parameter names to match the defaults defined in RemoteGateway.php? I'm not sure if the parameter bag is case-insenstive, but it would seem to make sense to be consistent with the other getters/setters...

coatesap commented on src/Omnipay/Realex/Message/SavedAuthRequest.php:
```diff
@@ -10,7 +10,7 @@
 class SavedAuthRequest extends RemoteAbstractRequest
 {
-    protected $endpoint = 'https://epage.payandshop.com/epage-remote.cgi';
+//    protected $endpoint = 'https://epage.payandshop.com/epage-remote.cgi';
```
If this line is no longer used, it should be removed rather than commented out.

coatesap commented on src/Omnipay/Realex/Message/AuthRequest.php:
```diff
@@ -10,7 +10,6 @@
  */
 class AuthRequest extends RemoteAbstractRequest
 {
-    protected $endpoint = 'https://epage.payandshop.com/epage-remote.cgi';
```
Would it not make more sense to leave the endpoint property in place, and allow the setter and getter to set it/access it? Also by removing it will break the existing Global Iris driver, which extends these requests (see https://github.com/coatesap/omnipay-global-iris/blob/master/src/Omnipay/GlobalIris/Message/AuthRequest.php for example). If there is no real reason to remove it, I'd rather not have to remove/upgrade/lock down the Global Iris driver...

lewisvgd replied:
I sort of agree with this implementation. This makes it possible to inject the parameters at the configuration stage (as per the merchant ID etc). Obviously the omnipay interfaces need to be adhered to so this results in not being able to explicitly pass the endpoint other than in the params.

3dgroup committed:
removed commented line

Commit:
xml text node coding problem bugfix

coatesap commented:
@James-Byrne - no, I think your implementation actually looks cleaner, so I'll probably merge yours instead. I'm guessing that your endpoint changes are to allow you to use it with white-labelled versions of Realex?

3dgroup commented:
@coatesap To me there is more than one end point URL and @James-Byrne commit only seems to cover the Auth URL not the data vault end point url.

coatesap commented:
@3dgroup - yes, good point. @James-Byrne - have you any thoughts how this could be addressed?

James-Byrne commented:
I think the pr by @3dgroup is far more comprehensive than the one I have made and covers my current use case as well as use cases I may have in the future. If you can resolve the outstanding issues you have yourself (styling, backwards compatibility etc) then I think this is the better option.

coatesap commented:
Thanks @James-Byrne, now that I've had chance to look properly at the commits, I'd agree. On that basis - @3dgroup - are you able to address the two comments I've added to the PR, and sort the indentation (tab is 4 spaces in Omnipay land, I believe), and I'll get this merged.

3dgroup commented:
I think the tab issue is from my college istvan when he resolved an XML issue regarding special characters. He will be updating this today.

Commit:
replace tabs to 4 spaces

Commit:
tidy changes

Commit:
RemoteAbstractRequest Amount node currency attribute value convert was wrong, move to convertation to node content

Commit:
added space after comma

3dgroup committed:
added Redirect features

3dgroup committed:
Merge remote-tracking branch 'refs/remotes/digitickets/master'

3dgroup committed:
format code

3dgroup committed:
cs fixer update

3dgroup committed:
Format fix's

3dgroup committed:
more formatting

3dgroup committed:
formatting issue

3dgroup committed:
formatting

3dgroup committed:
format issue

3dgroup committed:
Multi line format issue

3dgroup committed:
formatting

3dgroup committed:
formatting issue

3dgroup committed:
formatting last attempt

slogsdon commented:
What's the status of this PR? My team is working on Realex Hosted Payment Page / Redirect support as well, and it looks like some of the functionality will overlap what's being added here.

3dgroup commented:
My Fork is working fine, https://github.com/3dgroup/omnipay-realex

coatesap commented on nbproject/project.properties:
```diff
@@ -0,0 +1,7 @@
+include.path=${php.global.include.path}
```
Could you remove any of these IDE-specific files from the PR (I imagine they probably need to be gitignored...)

coatesap commented on src/Omnipay/Realex/Message/EnrolmentRequest.php:
```diff
@@ -120,8 +130,22 @@ protected function createResponse($data)
         return $response;
     }
 
+//    public function getEndpoint()
```
Commented-out code needs removing or a comment explaining why this is a temporary action adding.

coatesap reviewed (changes requested):
This looks very useful - but a couple of tweaks are needed before it can be merged - including the question about removing the protected endpoint properties.

3dgroup committed:
Code Tidy

3dgroup committed:
Merge remote-tracking branch 'refs/remotes/digitickets/master'

# Conflicts:
#	src/Omnipay/Realex/Message/RemoteAbstractRequest.php

3dgroup committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/Omnipay/Realex/Message/EnrolmentRequest.php
#	src/Omnipay/Realex/Message/RemoteAbstractRequest.php
#	src/Omnipay/Realex/RemoteGateway.php

3dgroup committed:
Rename response.php to Response.php

3dgroup committed:
Add script timeout

Elavon servers went down causing paying scripts to hang. Adding this frees the resources.

3dgroup committed:
Merge branch 'digitickets:master' into master

3dgroup committed:
Update dataVaultResponse.php

3dgroup committed:
Update RedirectCompleteStoreCardRequest.php

3dgroup committed:
Update RemoteAbstractRequest.php
